extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -232,11 +226,33 @@ ServiceStartBuildItem createBeansAndStartActions(FlywayRecorder recorder,
         }
     }
 
+    private String normalizeLocation(String location) {
+        if (location == null) {
+            throw new IllegalStateException("Flyway migration location may not be null.");
+        }
+
+        // Strip any 'classpath:' protocol prefixes because they are assumed
+        // but not recognized by ClassLoader.getResources()
+        if (location.startsWith(CLASSPATH_APPLICATION_MIGRATIONS_PROTOCOL + ':')) {
+            location = location.substring(CLASSPATH_APPLICATION_MIGRATIONS_PROTOCOL.length() + 1);
+            if (location.startsWith("/")) {
+                location = location.substring(1);
+            }
+        }
+        if (!location.endsWith("/")) {
+            location += "/";
+        }
+
+        return location;
+    }
+
     private Set<String> getApplicationMigrationsFromPath(final String location, final URL path)
             throws IOException, URISyntaxException {
-        try (final Stream<Path> pathStream = Files.walk(Paths.get(path.toURI()))) {
+        Path rootPath = Paths.get(path.toURI());
+
+        try (final Stream<Path> pathStream = Files.walk(rootPath)) {
             return pathStream.filter(Files::isRegularFile)
-                    .map(it -> Paths.get(location, it.getFileName().toString()).toString())
+                    .map(it -> Paths.get(location, rootPath.relativize(it).toString()).toString())
```

Comment:
This does not work in a Jar ! 
I get unable to optain inputstream for resource:
db/migration/oracle/../oracle/<migration_name>.sql

with quarkus/flyway.locations=db/migration/oracle

Reply:
I will try to make a reproducer yes

Reply:
Thanks

Reply:
Please open a separate issue with a reproducer so that we can easily track it.

Thanks!

Reply:
Please see #11780